react/features/recording/components/Recording/RecordButton.native.js

```diff
@@ -2,107 +2,19 @@
 
 import { connect } from 'react-redux';
 
-import { openDialog } from '../../../base/dialog';
 import { translate } from '../../../base/i18n';
-import { JitsiRecordingConstants } from '../../../base/lib-jitsi-meet';
-import {
-    isLocalParticipantModerator
-} from '../../../base/participants';
-import {
-    AbstractButton,
-    type AbstractButtonProps
-} from '../../../base/toolbox';
 
-import { getActiveSession } from '../../functions';
-
-import StartRecordingDialog from './StartRecordingDialog';
-import StopRecordingDialog from './StopRecordingDialog';
-
-/**
- * The type of the React {@code Component} props of {@link RecordButton}.
- */
-type Props = AbstractButtonProps & {
-
-    /**
-     * The current conference object.
-     */
-    _conference: Object,
-
-    /**
-     * True if there is a running active recording, false otherwise.
-     */
-    _isRecordingRunning: boolean,
-
-    /**
-     * The redux {@code dispatch} function.
-     */
-    dispatch: Function,
-
-    /**
-     * The i18n translate function.
-     */
-    t: Function
-};
+import AbstractRecordButton, {
+    _mapStateToProps,
+    type Props
+} from './AbstractRecordButton';
 
 /**
  * An implementation of a button for starting and stopping recording.
  */
-class RecordButton extends AbstractButton<Props, *> {
-    accessibilityLabel = 'Recording';
+class RecordButton extends AbstractRecordButton<Props> {
     iconName = 'recEnable';
-    label = 'dialog.startRecording';
     toggledIconName = 'recDisable';
```

Comment:
I wonder if we should just toggle the icon on web too. Maybe it'll be more clear to the user plus it'll move these classes into a shared component.

Reply:
Oh you already did do that. And I guess web was using a different icon the whole time.

Reply:
Yes, but I was a bit hesitant to change the web icon. What do you think?

Reply:
You or I should ask design. I assume mobile went through design as well and somehow got the icons that way, so I'm for consistency.

Reply:
Nah, you're too optimistic about available designs :) I just when with the most logical icon that had a toggled version too

![image](https://user-images.githubusercontent.com/9974975/42096198-f4b48f48-7bb4-11e8-876f-4a0da5d5b576.png)

But I go and ask for designs for sure

Reply:
I don't think so, Nicole seems to be busy and these new icons support more interaction than the old ones.

Reply:
👍

Reply:
@saghul Nicole asked me to use the old icon, so I updated it